File: common/src/main/java/au/com/codeka/warworlds/common/net/GzipHelper.java
package au.com.codeka.warworlds.common.net;

import com.google.common.io.ByteStreams;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.util.zip.GZIPInputStream;
import java.util.zip.GZIPOutputStream;

import au.com.codeka.warworlds.common.Log;

/**
 * Helper for working with Gzip compression.
 */
public class GzipHelper {
  private static final Log log = new Log("GzipHelper");

  /** Compress the given byte array. */
  public static byte[] compress(byte[] uncompressed) {
    try {
      ByteArrayOutputStream outs = new ByteArrayOutputStream();
      GZIPOutputStream zip = new GZIPOutputStream(outs);
      zip.write(uncompressed);
      zip.close();
      return outs.toByteArray();
    } catch (IOException e) {
      log.warning("Error compressing byte array.", e);
      return null;
    }
  }

  /** Decompress the given byte array. */
  public static byte[] decompress(byte[] compressed) {
    try {
      ByteArrayInputStream ins = new ByteArrayInputStream(compressed);
      GZIPInputStream zip = new GZIPInputStream(ins);
      return ByteStreams.toByteArray(zip);
    } catch (IOException e) {
      log.warning("Error decompressing byte array.", e);
      return null;
    }
  }
}

File: common/src/main/java/au/com/codeka/warworlds/common/net/GzipHelper.kt
package au.com.codeka.warworlds.common.net

import au.com.codeka.warworlds.common.Log
import com.google.common.io.ByteStreams
import java.io.ByteArrayInputStream
import java.io.ByteArrayOutputStream
import java.io.IOException
import java.util.zip.GZIPInputStream
import java.util.zip.GZIPOutputStream

/**
 * Helper for working with Gzip compression.
 */
object GzipHelper {
  private val log = Log("GzipHelper")

  /** Compress the given byte array.  */
  fun compress(uncompressed: ByteArray?): ByteArray? {
    return try {
      val outs = ByteArrayOutputStream()
      val zip = GZIPOutputStream(outs)
      zip.write(uncompressed)
      zip.close()
      outs.toByteArray()
    } catch (e: IOException) {
      log.warning("Error compressing byte array.", e)
      null
    }
  }

  /** Decompress the given byte array.  */
  fun decompress(compressed: ByteArray?): ByteArray? {
    return try {
      val ins = ByteArrayInputStream(compressed)
      val zip = GZIPInputStream(ins)
      ByteStreams.toByteArray(zip)
    } catch (e: IOException) {
      log.warning("Error decompressing byte array.", e)
      null
    }
  }
}